### PR TITLE
feat: add storage utilization gib per node for autoscaling

### PR DESCRIPTION
**Pull request by kolea2**

### CODE REVIEW

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableInstanceAdminClientIT.java [hunk 1]
```diff
@@ -343,6 +345,7 @@ public void createClusterWithAutoscalingAndPartialUpdateTest() {
                   ClusterAutoscalingConfig.of("ignored", clusterId)
                       .setMaxNodes(4)
                       .setMinNodes(1)
+                      .setStorageUtilizationGibPerNode(2561)
```
Comment (kongweihan) [inline google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableInstanceAdminClientIT.java:348]: Should we assert this number somewhere below?

**Review by kongweihan — APPROVED** [review]
I'm not very familiar with the java client code, but the changes themselves looks reasonable and good!
Had 1 comment in the test.

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableInstanceAdminClientIT.java (hunk 1 above)
Comment (kongweihan) [reply google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableInstanceAdminClientIT.java:348]: Oh I see you added it in another commit.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/ClusterAutoscalingConfig.java [hunk 2]
```diff
@@ -104,6 +104,21 @@ public ClusterAutoscalingConfig setCpuUtilizationTargetPercent(int cpuUtilizatio
...
+    builder.setUpdateMask(
+        FieldMaskUtil.union(
+            builder.getUpdateMask(),
+            FieldMaskUtil.fromString(
+                Cluster.class,
+                "cluster_config.cluster_autoscaling_config.autoscaling_targets.storage_utilization_gib_per_node")));
+    clusterConfigBuilder
+        .getClusterAutoscalingConfigBuilder()
+        .getAutoscalingTargetsBuilder()
+        .setStorageUtilizationGibPerNode(storageUtilizationGibPerNode);
```
Comment (mutianf) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/ClusterAutoscalingConfig.java:118]: Do we need to check if this is a valid value?

Comment (kolea2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/ClusterAutoscalingConfig.java:118]: The server will give back an `INVALID_ARGUMENT` error if the value is not within the range specified in the comments.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java [hunk 3]
```diff
@@ -116,6 +116,7 @@ public CreateClusterRequest setScalingMode(@Nonnull ClusterAutoscalingConfig aut
     int minNodes = autoscalingConfig.getMinNodes();
     int maxNodes = autoscalingConfig.getMaxNodes();
     int cpuTargetPercent = autoscalingConfig.getCpuUtilizationTargetPercent();
+    int storageUtilizationGibPerNode = autoscalingConfig.getStorageUtilizationGibPerNode();
```
[the remark this answers is not in the document]
Comment (kolea2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CreateClusterRequest.java:119]: good catch, done!

**Review by mutianf — APPROVED** [review]

**Pull request merged**